retrieval_service/datastore/providers/firestore.py

```diff
@@ -35,47 +38,111 @@ class Config(BaseModel, datastore.AbstractConfig):
 
 class Client(datastore.Client[Config]):
     __client: AsyncClient
+    policies_collection: AsyncQuery
+    amenities_collection: AsyncQuery
 
     @datastore.classproperty
     def kind(cls):
         return FIRESTORE_IDENTIFIER
 
     def __init__(self, client: AsyncClient):
         self.__client = client
+        self.policies_collection = AsyncQuery(self.__client.collection("policies"))
+        self.amenities_collection = AsyncQuery(self.__client.collection("amenities"))
 
     @classmethod
     async def create(cls, config: Config) -> "Client":
         return cls(AsyncClient(project=config.projectId))
 
+    async def delete_collections(collection_list: list[AsyncCollectionReference]):
+        # Checks if collection exists and deletes all documents
+        delete_tasks = []
+        for collection_ref in collection_list:
+            collection_exists = collection_ref.limit(1).stream()
+            if not collection_exists:
+                continue
+
+            docs = collection_ref.stream()
+            async for doc in docs:
+                delete_tasks.append(asyncio.create_task(doc.reference.delete()))
+        await asyncio.gather(*delete_tasks)
+
+    async def delete_indexes(index_list):
+        # Check if the collection exists and deletes all indexes
+        delete_tasks = []
+        for index in index_list:
+            if index:
+                delete_vector_index = [
+                    "gcloud",
+                    "alpha",
+                    "firestore",
+                    "indexes",
+                    "composite",
+                    "delete",
+                    index,
+                    "--database=(default)",
+                    "--quiet",  # Added to suppress delete warning
+                ]
+
+                delete_vector_index_process = await asyncio.create_subprocess_exec(
+                    *delete_vector_index,
+                )
```

Comment:
nit: why not the following
```suggestion
                delete_vector_index_process = await asyncio.create_subprocess_exec(
                    "gcloud",
                    "alpha",
                    "firestore",
                    "indexes",
                    "composite",
                    "delete",
                    index,
                    "--database=(default)",
                    "--quiet",  # Added to suppress delete warning
                )
```